dashboard/test/controllers/api_controller_test.rb

```diff
@@ -1600,4 +1600,68 @@ class ApiControllerTest < ActionController::TestCase
       {method: "get", path: "/api/student_progress/2/15"}
     )
   end
+
+  #
+  # Given two arrays, checks that they represent equivalent bags (or multisets)
+  # of elements.
+  #
+  # Equivalent:     [1, 1, 2], [1, 2, 1]
+  # Not equivalent: [1, 1, 2], [1, 2, 2]
+  #
+  # Optionally takes a comparator block.  If omitted, == comparison is used.
+  #
+  # equivalent_bags?([2, 3, 4], [12, 13, 14]) {|a,b| a%10 == b%10}
+  #
+  # @param [Array] bag_a
+  # @param [Array] bag_b
+  # @param [Block] (optional) comparator
+  # @return [Boolean] true if sets are equivalent, false if not
+  #
+  def equivalent_bags?(bag_a, bag_b)
+    bag_b_remaining = bag_b.clone
+    bag_a.each do |a|
+      match_index = bag_b_remaining.find_index do |b|
+        if block_given?
+          yield a, b
+        else
+          a == b
+        end
+      end
+      if match_index.nil?
+        return false
+      else
+        bag_b_remaining.delete_at match_index
+      end
+    end
+    bag_b_remaining.empty?
+  end
+
+  test 'equivalent_bags? helper' do
+    assert equivalent_bags? [], []
+    assert equivalent_bags? [1, 1, 1, 2, 2], [2, 1, 2, 1, 1]
+    refute equivalent_bags? [1, 1, 1, 2, 2], [1, 1, 2, 2, 2]
+    assert equivalent_bags? [2, 3, 4], [12, 13, 14] {|a, b| a % 10 == b % 10}
```

Comment:
@islemaster this will cause a syntax error starting in Ruby 2.5 (See [ruby#14023](https://bugs.ruby-lang.org/issues/14023) for discussion), and caused a sudden failure on my test branch (#20648).